plugin/sycl/objective/multiclass_obj.cc

```diff
@@ -119,65 +126,53 @@ class SoftmaxMultiClassObjOneAPI : public ObjFunction {
 
-
-    sycl::buffer<int, 1> additional_input_buf(1);
-	{
-		auto additional_input_acc = additional_input_buf.template get_access<sycl::access::mode::write>();
-		additional_input_acc[0] = 1; // Fill the label_correct flag
-	}
-
-
-    qu_.submit([&](sycl::handler& cgh) {
-      auto preds_acc            = preds_buf.template get_access<sycl::access::mode::read>(cgh);
-      auto labels_acc           = labels_buf.template get_access<sycl::access::mode::read>(cgh);
-      auto weights_acc          = weights_buf.template get_access<sycl::access::mode::read>(cgh);
-      auto out_gpair_acc        = out_gpair_buf.template get_access<sycl::access::mode::write>(cgh);
-      auto additional_input_acc = additional_input_buf.template get_access<sycl::access::mode::write>(cgh);
-      cgh.parallel_for<>(sycl::range<1>(ndata), [=](sycl::id<1> pid) {
-        int idx = pid[0];
-
-
-        bst_float const * point = &preds_acc[idx * nclass];
-
-
-        // Part of Softmax function
-        bst_float wmax = std::numeric_limits<bst_float>::min();
-        for (int k = 0; k < nclass; k++) { wmax = sycl::max(point[k], wmax); }
-        float wsum = 0.0f;
-        for (int k = 0; k < nclass; k++) { wsum += sycl::exp(point[k] - wmax); }
-        auto label = labels_acc[idx];
-        if (label < 0 || label >= nclass) {
-          additional_input_acc[0] = 0;
-          label = 0;
-        }
-        bst_float wt = is_null_weight ? 1.0f : weights_acc[idx];
-        for (int k = 0; k < nclass; ++k) {
-          bst_float p = expf(point[k] - wmax) / static_cast<float>(wsum);
-          const float eps = 1e-16f;
-          const bst_float h = sycl::max(2.0f * p * (1.0f - p) * wt, eps);
-          p = label == k ? p - 1.0f : p;
-          out_gpair_acc[idx * nclass + k] = GradientPair(p * wt, h);
-        }
-      });
-    }).wait();
-
-
     int flag = 1;
-	{
-		auto additional_input_acc = additional_input_buf.template get_access<sycl::access::mode::read>();
-		flag = additional_input_acc[0];
-	}
-
+    {
+      ::sycl::buffer<int, 1> additional_input_buf(&flag, 1);
+      qu_.submit([&](::sycl::handler& cgh) {
+        auto preds_acc            = preds_buf.template get_access<::sycl::access::mode::read>(cgh);
+        auto labels_acc           = labels_buf.template get_access<::sycl::access::mode::read>(cgh);
+        auto weights_acc          = weights_buf.template get_access<::sycl::access::mode::read>(cgh);
+        auto out_gpair_acc        = out_gpair_buf.template get_access<::sycl::access::mode::write>(cgh);
+        auto additional_input_acc = additional_input_buf.template get_access<::sycl::access::mode::write>(cgh);
+        cgh.parallel_for<>(::sycl::range<1>(ndata), [=](::sycl::id<1> pid) {
+          int idx = pid[0];
+
+
+          bst_float const * point = &preds_acc[idx * nclass];
+
+
+          // Part of Softmax function
+          bst_float wmax = std::numeric_limits<bst_float>::min();
```

Comment:
Worth a quick check here: https://www.intel.com/content/www/us/en/developer/articles/technical/the-maxloc-reduction-in-oneapi.html#gs.07t59o to make sure we are following fastest practice here. Maybe a nitpick, maybe not.  If I were thinking how I'd want this to be offloaded to a gpu, I'm thinking how the logic would convert over, and how it may try to optimize.